app/controllers/application_controller.rb

```diff
@@ -2,4 +2,21 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+
+  helper_method :current_user, :logged_in?
+
+  before_action :logged_in?
+
+  def current_user
+    User.find(session[:user_id]) if session[:user_id]
+  end
+
+
+  def logged_in?
```

Comment:
Calling a method a name ending in ? would lead a user to expect it to return a boolean an not have side effects. A better name might be redirct_unless_logged_in or similar.